### ISSUE TITLE
Create behavior concern tests

### ISSUE THREAD

**Issue by MadRabbit** [Prismatik/redbeard#152]
after the controllers and routers are uniformed and auto-generated, we can also start generating tests as well. something like this:

```js
const { app, itBehavesLikeCRUDResource } = require('../helper');

describe("/users route", () {
  itBehavesLikeCRUDResource(app);
});
```
eventually we could split those into specific concerns like, `itListsThingsProperly`, `itCreatesNewRecordsCorrectly`, etc.

**Comment by davidbanham** [comment]
Definitely interesting. I think we should do a little exploratory hacking to validate the approach. My concern is that it becomes difficult to inject things required by the business logic. Things like injecting auth credentials, etc.

**Comment by MadRabbit** [comment]
yup. absolutely, the point here is not to test custom business logic, but to test default _standard_ behavior of REST api that was generated by redbeard.

as for injecting the credentials, `doubleagent` has you covered:
```js
const { app, itBehavesLikeCRUDResource } = require('../helper');

describe("/users route", () {
  beforeEach(function * () {
    const user = yield new User({... }).save();
    const token = yield generateToken(user);

    app.defaultHeaders = { Authorization: `Bearer: ${token}` };
  });

  itBehavesLikeCRUDResource(app);
});
```

**Comment by davidbanham** [comment]
Ooooh, fancy. Yeah, looks like that'll work.
